File: labs/solidworks/swex/add-in/Overview.TaskPane.cs
public class TaskPaneControl : UserControl
{
}
public enum TaskPaneCommands_e
{
    Command1
}
    //...
    TaskPaneControl ctrl;
    var taskPaneView = CreateTaskPane<TaskPaneControl, TaskPaneCommands_e>(OnTaskPaneCommandClick, out ctrl);
    //...
private void OnTaskPaneCommandClick(TaskPaneCommands_e cmd)
{
    switch (cmd)
    {
        case TaskPaneCommands_e.Command1:
            //TODO: handle command
            break;
    }
}

File: labs/solidworks/swex/add-in/Overview.TaskPane.vb
Public Class TaskPaneControl
    Inherits UserControl
End Class
Public Enum TaskPaneCommands_e
    Command1
End Enum
    '...
    Dim ctrl As TaskPaneControl = Nothing
    Dim taskPaneView = CreateTaskPane(Of TaskPaneControl, TaskPaneCommands_e)(AddressOf OnTaskPaneCommandClick, ctrl)
    '...
Private Sub OnTaskPaneCommandClick(ByVal cmd As TaskPaneCommands_e)
    Select Case cmd
        Case TaskPaneCommands_e.Command1
    End Select
End Sub